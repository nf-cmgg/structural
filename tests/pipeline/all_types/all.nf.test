nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_all_types"

    options "-stub"

    test("all callers") {

        when {
            params {
                outdir = "${outputDir}"
                callers = "manta,smoove,delly,expansionhunter,qdnaseq,wisecondorx" //,gridss
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,

                    path("$outputDir")
                        .list()
                        .findAll {
                            !it.toString().endsWith("pipeline_info")
                        }
                        .collect {
                            it.list().collect { it.toString().split("/")[-1] }
                        },
                    path("$outputDir").list().collect{
                        it.toString().split("/")[-1]
                    }
                ).match("all")}
            )
        }

    }

    test("concat callers") {

        when {
            params {
                outdir = "${outputDir}"
                callers = "manta,smoove,delly,expansionhunter,qdnaseq,wisecondorx" //,gridss
                max_cpus   = 2
                max_memory = '6.GB'
                max_time   = '6.h'
                concat_output = true
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,

                    path("$outputDir")
                        .list()
                        .findAll {
                            !it.toString().endsWith("pipeline_info")
                        }
                        .collect {
                            it.list().collect { it.toString().split("/")[-1] }
                        },
                    path("$outputDir").list().collect{
                        it.toString().split("/")[-1]
                    }
                ).match("concat")}
            )
        }

    }

}
