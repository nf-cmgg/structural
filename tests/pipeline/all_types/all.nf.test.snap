{
    "all": {
        "content": [
            [
                
            ],
            [
                [
                    "PosCon1.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.qdnaseq.abberations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.qdnaseq.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.qdnaseq.cna:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.qdnaseq_segments.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "statistics.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    "PosCon2.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.qdnaseq.abberations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.qdnaseq.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.qdnaseq.cna:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.qdnaseq_segments.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "statistics.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    "PosCon3.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.qdnaseq.abberations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.qdnaseq.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.qdnaseq.cna:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.qdnaseq_segments.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "statistics.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    "PosCon4.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.qdnaseq.abberations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.qdnaseq.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.qdnaseq.cna:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.qdnaseq_segments.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "statistics.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    "PosCon5.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.qdnaseq.abberations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.qdnaseq.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.qdnaseq.cna:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.qdnaseq_segments.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "statistics.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "PosCon1",
                "PosCon2",
                "PosCon3",
                "PosCon4",
                "PosCon5",
                "pipeline_info"
            ]
        ],
        "timestamp": "2023-11-30T11:23:51.894310136"
    },
    "concat": {
        "content": [
            [
                
            ],
            [
                [
                    "PosCon1.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.concat.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.concat.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.qdnaseq.abberations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.qdnaseq.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.qdnaseq.cna:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.qdnaseq_segments.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon1.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "statistics.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    "PosCon2.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.concat.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.concat.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.qdnaseq.abberations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.qdnaseq.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.qdnaseq.cna:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.qdnaseq_segments.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon2.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "statistics.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    "PosCon3.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.concat.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.concat.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.qdnaseq.abberations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.qdnaseq.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.qdnaseq.cna:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.qdnaseq_segments.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon3.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "statistics.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    "PosCon4.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.concat.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.concat.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.qdnaseq.abberations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.qdnaseq.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.qdnaseq.cna:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.qdnaseq_segments.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon4.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "statistics.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                [
                    "PosCon5.cnv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.cnv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.concat.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.concat.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.qdnaseq.abberations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.qdnaseq.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.qdnaseq.cna:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.qdnaseq_segments.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.repeats.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.repeats.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.sv.vcf.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.sv.vcf.gz.tbi:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.wisecondorx_aberrations.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.wisecondorx_bins.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.wisecondorx_chr_statistics.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "PosCon5.wisecondorx_segments.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr1.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "chr22.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "genome_wide.png:md5,d41d8cd98f00b204e9800998ecf8427e",
                    "statistics.out:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            ],
            [
                "PosCon1",
                "PosCon2",
                "PosCon3",
                "PosCon4",
                "PosCon5",
                "pipeline_info"
            ]
        ],
        "timestamp": "2023-11-30T11:25:45.041210963"
    }
}